enhancements/image-references.md

```diff
@@ -0,0 +1,228 @@
+need to configure each build system independently.
+
+### Implementation Details/Notes/Constraints
+
+The core problems that must be solved in the implementation of this proposal have
+already been handled in the workflow used in `oc adm release new`. When implementing
+improvements to the `operator-sdk bundle create` process we will simply need to
+create a shared library for the two tools to use. While the shape of the output
+is slightly different and some of the semantics about how the output should be 
+formatted are dissmilar, the core image reference rewriting is identical and the
+process of building a `FROM scratch` image layer is also identical.
+
+Today, some prior art exists in the OSBS workflow for building operator bundles.
+As we improve the Operator SDK tooling to create a straghtforward process for
+creating bundle images, we must make sure a seamless migration is possible.
+
+### Risks and Mitigations
+
+It will be critical that the design be vetted by all of the concerned parties, from
+operator manifest authors to CI system authors and productized pipeline authors to
+ensure that the UX is appropriate in all cases. Furthermore, the largest risk in the
+implementation here is not prioritizing a clean migration pathway for all current
+users who create bundle images, which would lead to further fragmentation of the
+ecosystem, which is directly opposed to the goal of this enhancement.
+
+## Design Details
+
+The definition of a minimally-viable operator bundle image will be changed to
+ensure that all iamge references in the contained manifests have been resolved 
+to a digest and had the pull specifications rewritten to refer to those digests.
+
+The only acceptable process for creating an operator bundle will be to run the
+`operator-sdk bundle create` CLI, providing the manifests, metadata and image
+sources as input to the creation process.
+
+### Proposed UX
+
+Operator manifest authors write a manifest that refers to images using some
+opaque string [ex](https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_04_deployment.yaml#L22),
+and provide an `image-references` file alongside their manifests that declares
+which strings inside of their manifest are referring to pull specifications of
+images and names each occurence [ex](https://github.com/openshift/machine-config-operator/blob/98d9ba6841eb4811ed6f4d4de7016ea83c131c54/install/image-references#L7-L10).
+
+The `metadata/image-references` file holds data in the following format, mapping
+common names of container imags to their string placeholders in the manifests:
+
+```yaml
+imageReferences:
+- name: common-name
+  placeholder: registry.svc.ci.openshift.org/openshift:image
+```
+
+This mapping, therefore, defines what needs to be replced in manifests when
+they are bundled and identifies each replacement with a name. When
+`operator-sdk bundle create` runs, it will require as input a second mapping
```

Comment:
>as input

This would need to be a file/stored in a config somewhere, since typing all this out in an interactive prompt or in flag args would be painful.

Reply:
Yes, agreed.